Sprint10_proverka

VankaGam opened the pull request:
Отработали с наставником, разобрались с ветками.
Спасибо Игорю Авдонину!

Commit:
Сохранение изменений в gradle.xml

VankaGam committed:
Sprin10 ostalisResursi

VankaGam committed:
Sprin10 Gotov

VankaGam committed:
Sprin10 Gotov

VankaGam committed:
Завершено слияние с origin/main

VankaGam committed:
Sprin10

VankaGam committed:
ProverkaVetok

ArturNurtdinov commented on app/src/main/java/adapter/TrackAdapter.kt:
Класс необходимо перенести в пакет, где лежит весь остальной код - com.example.playlistmaker

ArturNurtdinov commented on app/src/main/java/adapter/TrackAdapter.kt:
```diff
@@ -0,0 +1,54 @@
+
+    override fun getFilter(): Filter {
+        TODO("Not yet implemented")
+    }
```
Интерфейс `Filterable` и эту функцию лучше убрать, т.к. они не нужны для поиска

ArturNurtdinov commented on app/src/main/java/model/TrackRepository.kt:
```diff
@@ -0,0 +1,13 @@
+package model
+
+class TrackRepository {
+    fun getSampleTracks(): ArrayList<Track> {
```
Рекомендуется использовать котлиновские интерфейсы `MutableList` и `List` для работы со списками. Создать такие списки можно с помощью функций `mutableListOf()` и `listOf()` соответственно

ArturNurtdinov commented on app/src/main/res/values/styleItem.xml:
```diff
@@ -0,0 +1,51 @@
+        <item name="android:layout_width">@dimen/layout_width_item</item>
+        <item name="android:layout_height">@dimen/layout_height_item</item>
+        <item name="android:layout_marginEnd">@dimen/margin_end_item</item>
+        <item name="background">@drawable/rounded_corners</item>
```
Здесь не применяется бэкграунд, т.к. свойство нужно писать с неймспейсом:
`android:background`

VankaGam committed:
Ispravil vse zamechaniya, takie kak:
-Zadniy plan, backround, ne sootvetstvoval dizauny;
-Zdelal zakrugleniya iconok treckov.

VankaGam committed:
Ispravil na: List<>

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/TrackAdapter.kt:
```diff
@@ -29,7 +29,7 @@ class TrackAdapter(private val tracks: List<Track>) : RecyclerView.Adapter<Track
                 .placeholder(R.drawable.placeholder)
                 .error(R.drawable.placeholder)
-                .centerCrop()
+                .transform(RoundedCorners(4))
```
В конструктор `RoundedCorners` нужно передавать кол-во пикселей. Это значит, что предварительно число, передаваемое в конструктор, нужно сконвертировать в пиксели, если мы хотим добиться одинаково скругления на девайсах с разной плотностью экрана
